msal/authority.py

```diff
@@ -140,8 +140,17 @@ def instance_discovery(url, http_client, **kwargs):
 def tenant_discovery(tenant_discovery_endpoint, http_client, **kwargs):
     # Returns Openid Configuration
     resp = http_client.get(tenant_discovery_endpoint, **kwargs)
-    payload = json.loads(resp.text)
-    if 'authorization_endpoint' in payload and 'token_endpoint' in payload:
-        return payload
-    raise MsalServiceError(status_code=resp.status_code, **payload)
+    if resp.status_code == 200:
+        payload = json.loads(resp.text)  # It could raise ValueError
+        if 'authorization_endpoint' in payload and 'token_endpoint' in payload:
+            return payload  # Happy path
+        raise ValueError("OIDC Discovery does not provide enough information")
+    if 400 <= resp.status_code < 500:
+        # Nonexist tenant would hit this path
+        # e.g. https://login.microsoftonline.com/nonexist_tenant/v2.0/.well-known/openid-configuration
+        raise ValueError("OIDC Discovery endpoint rejects our request")
```

Comment:
We can be a little more verbose on what needs to be done by the app. If this is hit when non existing tenant is specified, we can say specify an action item on app developer based on what needs to be done. I am not sure if the app developer will always understand what a OIDC discovery endpoint is and how is it formed since its an internal MSAL Implementation detail

Reply:
Good point. Actually, this exception will be [caught by ourselves at a different layer](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/269/files#diff-dea3df43cd06f684c7e95c0dcd265456fc0e2ed36cbdf7a72c6bc4dbd8af8f9aR86), and we provided information there.